pkg/controller/versions/versions.go

```diff
@@ -0,0 +1,167 @@
+package versions
+
+import (
+	"context"
+	"fmt"
+
+	v1 "github.com/maistra/istio-operator/pkg/apis/maistra/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// InvalidVersion is not a valid version
+	InvalidVersion version = iota
+	// UndefinedVersion is...undefined
+	UndefinedVersion
```

Comment:
The issue is that it means `DefaultVersion` if it's seen in the mutating webhook, and `LegacyVersion` in the processing.  I suppose mutating webhook could check for an empty version string instead of `UndefinedVersion`.

Reply:
Empty version now returns V1_0.  It was much easier to check the field for empty string in the mutating webhook than to have the code littered with `if version == versions.UndefinedVersion` all over the place.